File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOUpHandGestureParamModel.kt
//
//  IDOUpHandGestureParamModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Raise-to-wake gesture event number

open class IDOUpHandGestureParamModel(
    onOff: Int,
    showSecond: Int,
    hasTimeRange: Int,
    startHour: Int,
    startMinute: Int,
    endHour: Int,
    endMinute: Int
) : IDOBaseModel {

    /// Switch
    /// 1: On
    /// 0: Off                        
    @SerializedName("on_off")
    var onOff: Int = onOff

    /// Duration for the screen to stay on, in seconds
    @SerializedName("show_second")
    var showSecond: Int = showSecond

    /// Whether there is a time range
    /// 1: Yes
    /// 0: No 
    @SerializedName("has_time_range")
    var hasTimeRange: Int = hasTimeRange

    /// Starting hour of the time range
    @SerializedName("start_hour")
    var startHour: Int = startHour

    /// Starting minute of the time range
    @SerializedName("start_minute")
    var startMinute: Int = startMinute

    /// Ending hour of the time range
    @SerializedName("end_hour")
    var endHour: Int = endHour

    /// Ending minute of the time range
    @SerializedName("end_minute")
    var endMinute: Int = endMinute


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOUpHandGestureParamModel.swift
//
//  IDOUpHandGestureParamModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Raise-to-wake gesture event number
@objcMembers
public class IDOUpHandGestureParamModel: NSObject, IDOBaseModel {
    /// Switch
    /// 1: On
    /// 0: Off                        
    public var onOff: Int
    /// Duration for the screen to stay on, in seconds       
    public var showSecond: Int
    /// Whether there is a time range
    /// 1: Yes
    /// 0: No 
    public var hasTimeRange: Int
    /// Starting hour of the time range                      
    public var startHour: Int
    /// Starting minute of the time range                    
    public var startMinute: Int
    /// Ending hour of the time range                        
    public var endHour: Int
    /// Ending minute of the time range                      
    public var endMinute: Int
    
    enum CodingKeys: String, CodingKey {
        case onOff = "on_off"
        case showSecond = "show_second"
        case hasTimeRange = "has_time_range"
        case startHour = "start_hour"
        case startMinute = "start_minute"
        case endHour = "end_hour"
        case endMinute = "end_minute"
    }
    
    public init(onOff: Int,showSecond: Int,hasTimeRange: Int,startHour: Int,startMinute: Int,endHour: Int,endMinute: Int) {
        self.onOff = onOff
        self.showSecond = showSecond
        self.hasTimeRange = hasTimeRange
        self.startHour = startHour
        self.startMinute = startMinute
        self.endHour = endHour
        self.endMinute = endMinute
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

